pkg/listener/http2/http2_listener.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package http2
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+)
+
+import (
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+)
+
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/filterchain"
+	"github.com/apache/dubbo-go-pixiu/pkg/listener"
+	"github.com/apache/dubbo-go-pixiu/pkg/logger"
+	"github.com/apache/dubbo-go-pixiu/pkg/model"
+)
+
+func init() {
+	listener.SetListenerServiceFactory(model.ProtocolTypeHTTP2, newHttp2ListenerService)
+}
+
+type (
+	// ListenerService the facade of a listener
+	Http2ListenerService struct {
+		listener.BaseListenerService
+		listener net.Listener
+		server   *http.Server
+	}
+)
+
+type handleWrapper struct {
+	fc *filterchain.FilterChain
+}
+
+type h2cWrapper struct {
+	w *handleWrapper
+	h http.Handler
+}
+
+func (h *h2cWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.h.ServeHTTP(w, r)
+}
+
+// 转发 grpc 的请求
+func (h *handleWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.fc.ServeHTTP(w, r)
+}
+
+func newHttp2ListenerService(lc *model.Listener, bs *model.Bootstrap) (listener.ListenerService, error) {
+	fc := filterchain.CreateFilterChain(lc.FilterChain, bs)
+	return &Http2ListenerService{
+		BaseListenerService: listener.BaseListenerService{
+			Config:      lc,
+			FilterChain: fc,
+		},
+		listener: nil,
+		server:   nil,
+	}, nil
+}
+
+func (ls Http2ListenerService) Start() error {
```

Comment:
Add comment or ’nolint‘ for this func, pls.

Reply:
Done